### ISSUE TITLE
Not displayed on iOS 13.0 beta

### ISSUE THREAD

**Issue by yo1995** [hyperoslo/Lightbox#229]
Light box seems not to be presented properly on iOS 13.0 beta. More will be updated after another beta version of iOS is released next month.

The current behavior with Xcode 11 beta and iOS 13.0 beta is that the controller cannot be displayed when dynamic background is set to true.  
When dynamic background is set to false, though the controller can be presented, when trying to dismiss it either by tapping the close button, or by dragging up and down, it dismiss but the gesture seems to remain, which interrupts its parent's controls.

---

191219 Edit:

The current makeshift is to change it to `controller.modalPresentationStyle = .fullScreen`, to avoid the new `.automatic` style.

**Comment by victorlobe** [comment]
Please

**Comment by samyanez94** [comment]
+1

**Comment by kinoroy** [comment]
On iOS 13, the default `UIModalPresentationStyle` changed from `.fullScreen` to `.automatic`

To fix the issue where the controller isn't dismissed properly, change the `modalPresentationStyle` on `LightboxController` back to `UIModalPresentationStyle.fullScreen` like this:

```swift
 let controller = LightboxController(images: [LightboxImage(image: image)])
 controller.modalPresentationStyle = .fullScreen
 present(controller, animated: true)
```

**Comment by samyanez94** [comment]
Thanks @kinoroy. This solved the issue for me.

**Comment by smaljaar** [comment]
https://github.com/hyperoslo/Lightbox/issues/229#issuecomment-523589760 seemed initially to work, but I encountered another issue when presenting a LightboxController from another modally presented view controller. 

Steps to reproduce the issue:

1. present a modal X in the default iOS 13 `modalPresentationStyle` (`.automatic`, which means that on some devices the presenting controller is still visible in the background)
2. within this modal X I am presenting a `LightboxController` using the `modalPresentationStyle` set to `.fullScreen`. This results in an issue where the screen freezes and the `LightboxController` does not become visible. With the `LightboxController`'s `modalPresentationStyle`set to `.automatic` the original issue of this post is shown.

To work around the issue I am now presenting modal X also in `.fullScreen`

**Comment by jespercr** [comment]
Works as above for .fullScreen, but would love for it to work for .pageSheet/automatic as well in order to get consistency across the UI. Any update on this? Thanks a lot!

**Comment by zackshapiro** [comment]
I'm not getting anything presented when I set `modalPresentationStyle = .fullScreen`

**Comment by malonehedges** [comment]
@zackshapiro 

```swift
let lightbox = LightboxController(images: lightboxImages, startIndex: 0)
lightbox.modalPresentationStyle = .fullScreen
self.present(lightbox, animated: true)
```

**Comment by lawmaestro** [comment]
> I'm not getting anything presented when I set `modalPresentationStyle = .fullScreen`

Hard to tell without more context but I suspect in your case, the reason you're not seeing anything show when using `fullScreen` is that you already have a modally presented view controller. For example, I was using `Gallery` to display thumbnails and then `Lightbox` to preview these. The underlying Gallery controller was already presented modally which causes the modal presentation of the `Lightbox` on top of that to fail if using `fullScreen`.

So in summary, using `fullscreen` isn't ideal for the reasons of inconsistent presentation style (if you've adopted the new style elsewhere) and it won't work if you've already got a modally presented controller :(

**Comment by gowtham171** [comment]
> @zackshapiro
> 
> ```swift
> let lightbox = LightboxController(images: lightboxImages, startIndex: 0)
> lightbox.modalPresentationStyle = .fullScreen
> self.present(lightbox, animated: true)
> ```

which is already presenting (null) Im getting this error  @malonehedges

**Comment by benSmith1981** [comment]
When I press close on lightbox it does not dismiss the images, even if i present it full screen

**Comment by 3lvis** [comment]
@benSmith1981 can you open a separte issue for this, please? Are you able to reproduce this in the demo project?
